apps/balancer-tools/src/app/apr/(utils)/fetchPoolSnapshots.ts

```diff
@@ -0,0 +1,43 @@
+import { PoolSnapshotInRangeQuery } from "@bleu-balancer-tools/gql/src/balancer/__generated__/Ethereum.server";
+
+import { pools, poolsWithoutCache } from "#/lib/gql/server";
+
+import { dateToEpoch, generateDateRange } from "../api/(utils)/date";
+
+export async function fetchPoolSnapshots(
+  to: number,
+  extendedFrom: number,
+  network: string,
+  poolId: string,
+): Promise<PoolSnapshotInRangeQuery> {
+  let res: PoolSnapshotInRangeQuery;
+
+  const now = new Date();
+  const currentDateUTC = dateToEpoch(
+    new Date(
+      Date.UTC(
+        now.getUTCFullYear(),
+        now.getUTCMonth(),
+        now.getUTCDate(),
+        0,
+        0,
+        0,
+        0,
+      ),
+    ),
+  );
+
+  if (to === currentDateUTC) {
+    res = await poolsWithoutCache.gql(network).poolSnapshotInRange({
+      poolId,
+      timestamp: generateDateRange(extendedFrom, to),
+    });
+  } else {
+    res = await pools.gql(network).poolSnapshotInRange({
+      poolId,
+      timestamp: generateDateRange(extendedFrom, to),
```

Comment:
I'm not sure if I said this, but I wonder if we're having errors because we're filtering using timestamps we generated in a list instead of more loosely using `timestamp_gte` and `timestamp_lte` filters. I thought that was what we were using but could be remembering this wrong. Do you know why we're not using that approach?

Reply:
IMO they both do the same, just different versions to achieve the same thing. This a change during the refactor you made https://github.com/bleu-fi/balancer-tools/blob/81745fb789394fe760ac2de431d50bd7c74e8c11/packages/gql/src/balancer/pools.ts and I thought it was well put as the timestamps are specific